src/state/comments.ts

```diff
@@ -1,330 +1,291 @@
 import type { Annotation } from '../utils/annotation';
 import * as actions from '../actions/comments';
-
-type Partial<T> = {
-    [P in keyof T]?: T[P];
-};
+import { Partial, update } from './utils';
 
 export interface Author {
-    id: any;
-    name: string;
+  id: any;
+  name: string;
 }
 
 export type CommentReplyMode =
-    | 'default'
-    | 'editing'
-    | 'saving'
-    | 'delete_confirm'
-    | 'deleting'
-    | 'deleted'
-    | 'save_error'
-    | 'delete_error';
+  | 'default'
+  | 'editing'
+  | 'saving'
+  | 'delete_confirm'
+  | 'deleting'
+  | 'deleted'
+  | 'save_error'
+  | 'delete_error';
 
 export interface CommentReply {
-    localId: number;
-    remoteId: number | null;
-    mode: CommentReplyMode;
-    author: Author | null;
-    date: number;
-    text: string;
-    editPreviousText: string;
-    deleted: boolean;
+  localId: number;
+  remoteId: number | null;
+  mode: CommentReplyMode;
+  author: Author | null;
+  date: number;
+  text: string;
+  editPreviousText: string;
+  deleted: boolean;
 }
 
 export interface NewReplyOptions {
-    remoteId?: number | null;
-    mode?: CommentReplyMode;
-    text?: string;
+  remoteId?: number | null;
+  mode?: CommentReplyMode;
+  text?: string;
 }
 
 export function newCommentReply(
-    localId: number,
-    author: Author | null,
-    date: number,
-    {
-        remoteId = <number | null>null,
-        mode = <CommentReplyMode>'default',
-        text = ''
-    }: NewReplyOptions
+  localId: number,
+  author: Author | null,
+  date: number,
+  {
+    remoteId = <number | null>null,
+    mode = <CommentReplyMode>'default',
```

Comment:
**Issue:** you shouldn't need to cast literals in the first place....  Also, I suggest you stick to one casting type for consistency's sake. Unless you can't use it in this context? I like`as Type` more since the angle bracket syntax is also used for generics. There's no preference according to TypeScript docs - https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions

Reply:
Okay, removed now, thanks. I think this weirdly might have been causing some of my other problems, because as soon as I removed this, I was able to clear up the other errors that I was using casting to get around in the tests